modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/TableKeyDetails/index.js

```diff
@@ -0,0 +1,308 @@
+								/>
+							)}
+						</p>
+					</div>
+
+					<div className="col-3">
+						<p className="bg-neutral-1 key-details-paragraph px-3 py-2 rounded">
+							{getCurrentEndDate(activationKeys.createDate)}
+						</p>
+					</div>
+				</div>
+
+				<div
+					className={classNames('row', {
+						'justify-content-between': hasVirtualCluster,
+					})}
+				>
+					<div className="col-5">
+						<p className="text-neutral-8 text-paragraph-sm">
+							Instance Size
+						</p>
+					</div>
+
+					{!hasVirtualCluster && (
+						<div className="col-4">
+							<p className="text-neutral-8 text-paragraph-sm">
+								{IP_ADDRESSES}
+							</p>
+						</div>
+					)}
+
+					<div className="col-3">
+						<p className="text-neutral-8 text-paragraph-sm">
+							Expiration Date
+						</p>
+					</div>
+				</div>
+
+				<div
+					className={classNames('row', {
+						'justify-content-between': hasVirtualCluster,
+					})}
+				>
+					<div className="col-5">
+						<p className="bg-brand-primary-lighten-5 key-details-paragraph px-3 py-2 rounded">
+							{instanceSizeFormated}
+						</p>
+					</div>
+
+					{!hasVirtualCluster && (
+						<div className="col-4">
+							<p className="bg-neutral-1 d-flex key-details-paragraph px-3 py-2 rounded">
+								{activationKeys.ipAddresses || '-'}
+
+								{activationKeys.ipAddresses && (
+									<ClayIcon
+										className="copy-clipboard-icon ml-3 mt-1 text-neutral-5"
+										onClick={() =>
+											handleCopyToClipboard(
+												IP_ADDRESSES,
+												activationKeys.ipAddresses
+											)
+										}
+										symbol="copy"
+									/>
+								)}
+							</p>
+						</div>
+					)}
+
+					<div className="col-3">
+						<p className="bg-neutral-1 key-details-paragraph px-3 py-2 rounded">
+							{new Date(activationKeys.expirationDate) >=
+							new Date(unlimitedLicenseDate)
+								? 'Does Not Expire'
+								: getCurrentEndDate(
+										activationKeys.expirationDate
+								  )}
+						</p>
+					</div>
+				</div>
+
+				{!hasVirtualCluster && (
+					<>
+						<div className="justify-content-center row">
+							<div className="col-2">
+								<p className="text-neutral-8 text-paragraph-sm">
+									{MAC_ADDRESSES}
+								</p>
+							</div>
+						</div>
+						<div className="justify-content-center row">
+							<div className="col-4 ml-8">
+								<p className="bg-neutral-1 d-flex key-details-paragraph px-3 py-2 rounded">
+									{activationKeys.macAddresses || '-'}
+
+									{activationKeys.macAddresses && (
+										<ClayIcon
+											className="copy-clipboard-icon ml-3 mt-1 text-neutral-5"
+											onClick={() =>
+												handleCopyToClipboard(
+													MAC_ADDRESSES,
+													activationKeys.macAddresses
+												)
+											}
+											symbol="copy"
+										/>
+									)}
+								</p>
+							</div>
+						</div>
+					</>
+				)}
+			</div>
+		</>
+	);
+};
+export default TableKeyDetails;
```

Comment:
So, I saw that we have a lot of already made logic in others files. I also know that they're not available to use yet (they're not being exported). I suggest you create in DXPActivationKeysTable/utils/constants the files needed to handle those validations that I pointed here in this TableKeyDetails component.

If you need any help to a better understanding please let me know

Reply:
Nice, already done! Thank you for your observations Iago.